kernel/src/hil/crc.rs

```diff
@@ -0,0 +1,35 @@
+// Generic interface for CRC computation
+
+use returncode::ReturnCode;
+
+#[derive(Copy, Clone)]
+pub enum Polynomial {
+	CCIT8023,   // Polynomial 0x04C11DB7
+	CASTAGNOLI, // Polynomial 0x1EDC6F41
+	CCIT16,     // Polynomial 0x1021
+}
+
+pub fn poly_from_int(i: usize) -> Option<Polynomial> {
+    match i {
+        0 => Some(Polynomial::CCIT8023),
+        1 => Some(Polynomial::CASTAGNOLI),
+        2 => Some(Polynomial::CCIT16),
+        _ => None
+    }
+}
+
+pub trait CRC {
+    // Get the version of the CRC unit
+    fn get_version(&self) -> u32;
```

Comment:
Something akin to `get_version` that's likely important is a way for software to query what types of CRCs the underlying hardware supports (as you noted above, there's a bunch of options). In practice, that's what the CRC version is doing for the SAM4L, but the version <> feature mapping is left to the SAM4L datasheet and user code. It would be great for that mapping to be handled by Tock in a more structured manner